Update localization plugin documentation (#1602)
google/automotive-design-compose: docs/_docs/localization.md

    can download it from the [Figma downloads page][2].

4.  Import the plugin from manifest and you are ready to run the plugin. The file path is:

    ```text
    automotive-design-compose/support-figma/extended-layout-plugin/manifest.json
    ```

    <img src="./localization/ImportPluginFromManifest.png">

5.  Run the plugin under development through:
    **Figma Main Menu -> Plugins -> Development**
    **-> Automotive Design For Compose -> Localization Tool -> Generate String Resource**

### 1.2 Strings generation options

Currently the plugin provides three options to exclude text nodes from translations.

<img src="./localization/LocalizationOptions.png">

The first option excludes text nodes whose name starts with `#` which is a naming convention for text
nodes designed for app customization.

To use the second option, you will need to run the **Check/update keywords** plugin first to upload
the generated json file under your android app's build folder. The keywords json file contains app
customizations for nodes so the localization plugin can exclude those text nodes from translation.

> It also use the top level component names from the keywords file to locate frames that will be rendered

Reviewer: ```suggestion
It also uses the top level component names from the keywords file to locate frames that will be rendered
```

Author: Done.

on the client side and will only pull strings from the text nodes in those frames. By using the keywords
file, it will exclude text nodes that are used as comments, for example.

<img src="./localization/CheckOrUpdateKeywords.png">

The third option, checked by default, will generate one entry in the generated strings xml file for each
identical string. By unchecking it, it will generate one entry for each string instance no matter if it

> is the same with some of the other text nodes. Later on, you can review and manually merge same strings

Reviewer: ```suggestion
is the same with some of the other text nodes. Later on, you can review and manually merge the same strings
```

Author: Done.

The plugin also allows you to set a proper description and char limit for translations.

#### 1.4.1 Break identical strings into different translations

> Below is an exmaple of the output strings with the group option checked.

Reviewer: ```suggestion
Below is an example of the output strings with the group option checked.
```

Author: Done.

<img src="./localization/GroupedExample.png">

The strings grouping several text nodes will have a break icon button for each of the node, by clicking on which,
it will create a separate entry for that node with a different string resource id. The string resource id is auto
generated by the plugin. You can rename it. Note, if you rename it to an exising string resource name with the

> same text, it will merge them together.

Reviewer: Hard to suggest over multiple lines, so:
"The strings grouping several text nodes will have a break icon button for each of node. By clicking this icon,
a separate entry for that node will be created with a different string resource id. The string resource id is auto
generated by the plugin, but you can rename it. Note, if you rename it to an existing string resource name with the same text, it will merge them together."

Author: Done.

<img src="./localization/UngroupedExample.png">

> The same, when it comes to the output with the group option unchecked, you can merge them by renaming them to use the same string resource name.

Reviewer: I'm not sure what "The same" is referring to at the beginning of this sentence -- maybe leave it out, unless I'm misunderstanding.

Author: Done.

We are working on a few improvements for the plugin:

> 1. The user action to break identital string into different translations are not saved. We are looking into

Reviewer: ```suggestion
1. The user action to break identical strings into different translations is not saved. We are looking into
```

Author: Done.
